lambdaclass/mirra_backend
# [GH-565] add trophies currency

## Motivation
When a match ends you should receive or loss x amount of trophies depending on which positions you ended in the match and your current amount of trophies

Closes #565

the levels and amounts especification can be found here: https://www.notion.so/lambdaclass/Ranking-System-c04164a466b64e27ba9f9c8e81d755c6

## Summary of changes
- Added position check to the match result struct
- Now every time a player dies the result will be stored in a map 

## How to test it?

1. Create a google user with: `{:ok, user} = GameBackend.Users.find_or_create_google_user_by_email("a")`
2. use the user id generated in 1 and play a match with the following link in the browser: `http://localhost:3000/board/YUOR_USER_ID/uma/jorge/join`
3. When the match the user that you created should receive x amount of trophies depending in how many you had previously and your position. The rules for this can be found in `apps/game_backend/priv/currencies_rules.json` and you can check how many trophies you have with the following command: `Repo.get_by(GameBackend.Users.GoogleUser, id: YOUR_USER_ID ) |> Repo.preload([user: [currencies: :currency]])`

## Checklist
